rcl/src/rcl/wait.c

```diff
@@ -64,6 +64,7 @@ static void
 __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
 {
   rcl_ret_t ret;
+  (void)ret;  // NO LINT
```

Comment:
Why not declare the `ret` variable in each local scope since the variable seems to never be used in this bigger scope?

Also shouldn't all the checks with `assert` also be done when building release?